### ISSUE TITLE
Fix periodic vercel build errors

### ISSUE THREAD

**Issue by alalonde** [MetaFam/TheGame#508]
e.g.

```
@metafam/web: Error occurred prerendering page "/player/mlibty". Read more: https://nextjs.org/docs/messages/prerender-error
--
23:44:37.765 | @metafam/web: Error: `redirect` can not be returned from getStaticProps during prerendering (/player/mlibty)
23:44:37.765 | @metafam/web: See more info here: https://nextjs.org/docs/messages/gsp-redirect-during-prerender
```

**Comment by alalonde** [comment]
This may be [quite easy](https://nextjs.org/docs/messages/gsp-redirect-during-prerender) but difficult to test...

**Comment by dysbulic** [comment]
My understanding is the redirect is being returned thanks to [this code](https://github.com/MetaFam/TheGame/blob/dfff04ebaaa21b82403df9c5f4f1cc9361073eca/packages/web/pages/player/%5Busername%5D.tsx#L197-L208) which attempts to look up a user, and, if it isn't found, attempts to look up the lowercased version of the name. When that is found, it redirects to the lowercase url.

The username in question, "mlibty", is already lowercased, but I think the first lookup fails due to a backend timeout.

I'm pretty sure the solution is to [add retries to urql](https://formidable.com/open-source/urql/docs/advanced/retry-operations/) ([here](https://github.com/MetaFam/TheGame/blob/dfff04ebaaa21b82403df9c5f4f1cc9361073eca/packages/web/graphql/client.ts)) so that when it times out it doesn't return an error, but instead waits and tries again.

If it is a timeout error, setting a [fallback page](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages) won't necessarily fix the problem since it could timeout twice in a row. This is less likely, so the error will occur less, but it doesn't address the fundamental issue.

**Comment by alalonde** [comment]
> The username in question, "mlibty", is already lowercased, but I think the first lookup fails due to a backend timeout.

Just curious, have you verified this in the logs?

**Comment by dysbulic** [comment]
There's not really anything in the logs other than the failure. I have seen timeouts when accessing Hasura.

It would explain the intermittent nature of the errors.

**Comment by dan13ram** [comment]
I've verified that Timeout is indeed the issue. Added retries in #548 
But it still fails sometimes. 

Looks like even 2 retries are not enough sometimes.

Perhaps worth thinking about doing some load balancing on hasura.
https://hasura.io/learn/graphql/hasura-advanced/performance/2-horizontal-scaling/

**Comment by hammadj** [comment]
It might just be best to migrate to Hasura Cloud

**Comment by dan13ram** [comment]
You are correct Hammad. The Github Actions errors shouldn't be a problem anymore thought. 
This issue is regarding the builds failing on Vercel.

I think moving to Hasura Cloud is the best option too.

**Comment by alalonde** [comment]
#482 is the ticket regarding the Github Actions build.  Should be fixed now.  Do we know what the resource constraints are with Hasura? Migrating it to Cloud could be a big lift as it entails upgrading to 2.0 which I believe is not even in prerelease yet. I would be more comfortable throwing more resources at our instance to address the issue, at least in the short-term.

**Comment by alalonde** [comment]
> Don't know what the constraints are.
> Just know that it throws up ResponseTimeout as a GraphQLError. Difficult to judge where this is coming from.
> After our recent call, I'm wondering if 3box could be an issue as well. In every call for players or patrons, hasura in turn calls 3box. Which might potentially be leading to these timeouts!

Right. So migrating to Cloud may or may not fix the issue as we don't actually know what's causing it! I will be deploying a staging instance soon which we can point Vercel to rather than production. During static page generation (players and patrons now) it's probably hitting Hasura pretty hard so I would add some logging to those endpoints as a starter.

**Comment by dan13ram** [comment]
Something else worth a try 
https://hasura.io/docs/latest/graphql/cloud/response-caching.html
